src/rdkafka_metadata_cache.c

```diff
@@ -221,6 +209,25 @@ rd_kafka_metadata_cache_find(rd_kafka_t *rk, const char *topic, int valid) {
         return NULL;
 }
 
+/**
+ * @brief Find cache entry by topic id
+ *
+ * @param valid: entry must be valid (not hint)
+ *
+ * @locks rd_kafka_*lock()
+ */
+struct rd_kafka_metadata_cache_entry *
+rd_kafka_metadata_cache_find_by_id(rd_kafka_t *rk,
+                                   const rd_kafka_Uuid_t topic_id,
```

Comment:
`const rd_kafka_Uuid_t` passes "by const value," which is never correct. Surely the compiler warned about this?

Everywhere else in librdkafka that takes `rd_kafka_Uuid_t` either takes it by pointer to avoid a copy — `const rd_kafka_Uuid_t*` — which is probably what you want here, since you never use the parameter's own value and it's large — or takes it by "non-const value" in two places (`rd_kafka_topic_partition_new_with_topic_id` and `rd_kafka_Uuid_cmp`: the latter is inline and the former looks like a flaw to me but it's part of the public API now).

Reply:
Internally we pass the Uuid by struct copy, as a primitive value, to avoid more heap allocations. Here the const isn't strictly necessary, but it doesn't hurt as that parameter shouldn't be modified inside the function